File: src/com/fusenetworks/fuse/commands/Command_launchpads.java
package com.fusenetworks.fuse.commands;

import com.fusenetworks.fuse.util.NUtil;
import org.bukkit.command.Command;
import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;
import org.bukkit.ChatColor;

@CommandPermissions(source = SourceType.BOTH)
@CommandParameters(description = "Enable or disable launchpads", usage = "/<command> [on | off | status]")
public class Command_launchpads extends BaseCommand {
    @Override
    public boolean run(final CommandSender sender, final Player sender_p, final Command cmd, final String commandLabel, final String[] args, final boolean senderIsConsole) 
    {
        String enabled = plugin.getConfig().getString("launchpads.enabled");
        
        if (!sender.hasPermission("fuse.launchpads"))
        {
            sender.sendMessage(Messages.MSG_NO_PERMS);
            return true;
        }

        if (args.length != 1)
        {
            return false;
        }
        
        if (args[0].equalsIgnoreCase("enable") || args[0].equalsIgnoreCase("on"))
        {
            NUtil.adminAction(sender.getName(), "Enabling launchpads", false);
            plugin.getConfig().set("launchpads.enabled", "true");
            plugin.saveConfig();
            plugin.reloadConfig();
            return true;
        }
        if (args[0].equalsIgnoreCase("disable") || args[0].equalsIgnoreCase("off"))
        {
            NUtil.adminAction(sender.getName(), "Disabling launchpads", false);
            plugin.getConfig().set("launchpads.enabled", "false");
            plugin.saveConfig();
            plugin.reloadConfig();
            return true;
        }
        if (args[0].equalsIgnoreCase("status"))
        {
            sender.sendMessage(ChatColor.BLUE + "Launchpads enabled: " + enabled);
            return true;
        }
        return true;
    }
}

File: src/com/fusenetworks/fuse/commands/Command_launchpads.kt
package com.fusenetworks.fuse.commands

import com.fusenetworks.fuse.util.NUtil
import org.bukkit.Bukkit
import org.bukkit.ChatColor
import org.bukkit.command.Command
import org.bukkit.command.CommandExecutor
import org.bukkit.command.CommandSender
import org.bukkit.entity.Player

abstract class Command_launchpads : CommandExecutor {
    fun run(sender: CommandSender, sender_p: Player, cmd: Command, commandLabel: String, args: Array<String>, senderIsConsole: Boolean): Boolean {
        val enabled = Bukkit.getPluginManager().getPlugin("Fuse").config.getString("launchpads.enabled")

        if (!sender.hasPermission("fuse.launchpads")) {
            sender.sendMessage(Messages.MSG_NO_PERMS)
            return true
        }

        if (args.size != 1) {
            return false
        }

        if (args[0].equals("enable", ignoreCase = true) || args[0].equals("on", ignoreCase = true)) {
            NUtil.adminAction(sender.name, "Enabling launchpads", false)
            Bukkit.getPluginManager().getPlugin("Fuse").config.set("launchpads.enabled", "true")
            Bukkit.getPluginManager().getPlugin("Fuse").saveConfig()
            Bukkit.getPluginManager().getPlugin("Fuse").reloadConfig()
            return true
        }
        if (args[0].equals("disable", ignoreCase = true) || args[0].equals("off", ignoreCase = true)) {
            NUtil.adminAction(sender.name, "Disabling launchpads", false)
            Bukkit.getPluginManager().getPlugin("Fuse").config.set("launchpads.enabled", "false")
            Bukkit.getPluginManager().getPlugin("Fuse").saveConfig()
            Bukkit.getPluginManager().getPlugin("Fuse").reloadConfig()
            return true
        }
        if (args[0].equals("status", ignoreCase = true)) {
            sender.sendMessage(ChatColor.BLUE.toString() + "Launchpads enabled: " + enabled)
            return true
        }
        return true
    }
}
